web/src/hooks/useTodoAPI.tsx

```diff
@@ -10,5 +12,56 @@ export function useTodoAPI() {
     setTodoAPI(new TodoAPI(token))
   }, [token])
 
-  return todoAPI
+  const addTodo = useCallback(
+    (title: string) => {
+      const todo = todoAPI
+        .create({ addTodoRequest: { title: title } })
+        .then((response) => response.data)
+        .catch((error: AxiosError<ErrorResponse>) => {
+          throw new Error(error.message)
+        })
+      return todo
+    },
+    [todoAPI]
+  )
+
+  const getAllTodos = useCallback(() => {
+    const todos = todoAPI
+      .getAll()
+      .then((response) => response.data)
+      .catch((error: AxiosError<ErrorResponse>) => {
+        throw new Error(error.message)
+      })
+    return todos
+  }, [todoAPI])
+
+  const toggleTodo = useCallback(
+    async (todo: AddTodoResponse) => {
+      todoAPI
+        .updateById({
+          id: todo.id,
+          updateTodoRequest: {
+            is_completed: !todo.is_completed,
+            title: todo.title,
+          },
+        })
+        .catch((error: AxiosError<ErrorResponse>) => {
+          throw new Error(error.message)
+        })
+    },
+    [todoAPI]
+  )
+
+  const removeTodo = useCallback(
+    async (todo: AddTodoResponse) => {
+      todoAPI
+        .deleteById({ id: todo.id })
+        .catch((error: AxiosError<ErrorResponse>) => {
+          throw new Error(error.message)
+        })
```

Comment:
This one is repeated 4 times. Perhaps refactor?

Reply:
Thank you! And yeah, good observed! 😸 It was repeated just as many times up to this change, though. It's a lot easier to find patterns like this now, as we've collected the functions in a common parent and gotten rid of some unwanted dependencies. I think the next step would be to write a generic fetch function. 👍

Reply:
Follow up in #200